### PR TITLE
Grade case insensitive

### LINKED ISSUE #126: AddGrade case sensitivity

### PR DESCRIPTION
**Pull request by dominopizzaaaa**
Fixes #126

### CODE REVIEW

File: src/main/java/seedu/address/model/person/GradeList.java [hunk 1]
```diff
@@ -175,7 +187,14 @@ public boolean equals(Object other) {
         }
 
         GradeList otherGradeList = (GradeList) other;
-        return this.grades.equals(otherGradeList.grades);
+        // Sort both grade lists before comparing
+        List<Grade> sortedGrades = new ArrayList<>(this.grades.values());
+        sortedGrades.sort(Comparator.comparing(Grade::getTestName));
+
+        List<Grade> otherSortedGrades = new ArrayList<>(otherGradeList.grades.values());
+        otherSortedGrades.sort(Comparator.comparing(Grade::getTestName));
```
Comment (juancarlovieri) [inline src/main/java/seedu/address/model/person/GradeList.java:195]: simply comparing does not work. For example:

`[a, B]`
and
`[A,b]`

Instead, you can try to convert everything to lower case first before sorting and comparing.

**Review by juancarlovieri — CHANGES_REQUESTED** [review]

**Review by juancarlovieri — DISMISSED** [review]
lgtm good work wow

**Review by dominopizzaaaa — COMMENTED** [review]
lgtm (ig)

**Review by izruff — CHANGES_REQUESTED** [review]
`deleteGrade` cannot detect same test names with different cases, e.g. here `deleteGrade 1 n/test` works as expected (deletes 'Test') but `deleteGrade 1 n/Test` does not
image

Otherwise, `addGrade` and `aggGrade` seems to be working fine.

The problem is `removeGrade` and `getGrade` do not normalize the test name before searching. And `deleteGradeCommand` internally calls `getGrade` to check for test name, hence the issue.

So the best fix I think is to make `getGrade` directly convert test name to lowercase.

```
    public Grade getGrade(String testName) {
        requireNonNull(testName);
        String normalizedTestName = testName.toLowerCase();
        return this.grades.get(normalizedTestName);
    }
```

File: src/test/java/seedu/address/model/person/GradeListTest.java [hunk 2]
```diff
@@ -56,12 +63,12 @@ public void getGrade() {
...
-        assertEquals(gradeList.getGrade("Midterm"), firstGrade);
-        assertEquals(gradeList.getGrade("Final"), secondGrade);
-        assertEquals(gradeList.getGrade("Assignment"), thirdGrade);
+        assertEquals(gradeList.getGrade("midterm"), firstGrade);
+        assertEquals(gradeList.getGrade("final"), secondGrade);
+        assertEquals(gradeList.getGrade("assignment"), thirdGrade);
 
         // non-existing test name -> returns null
-        assertNull(gradeList.getGrade("Tutorial"));
+        assertNull(gradeList.getGrade("tutorial"));
```
Comment (izruff) [inline src/test/java/seedu/address/model/person/GradeListTest.java:71]: After modifying `getGrade` as mentioned, let's update the test cases to include variations of capital cases like "Midterm" and "MIDTERM".

**Review by izruff — APPROVED** [review]
Nice work!

**Pull request merged**